scripts/release/build-experimental-typescript.js

```diff
@@ -19,5 +19,11 @@ fs.writeFileSync(
   `export default '${newVersion}';\n`
 );
 
-shell.exec(`NODE_ENV=production VERSION=${newVersion} yarn build`);
-shell.exec('yarn build:types');
+if (
+  shell.exec(`NODE_ENV=production VERSION=${newVersion} yarn build`).code !== 0
+) {
+  shell.exit(1);
+}
+if (shell.exec('yarn build:types').code !== 0) {
+  shell.exit(1);
+}
```

Comment:
```suggestion
shell.exec(`NODE_ENV=production VERSION=${newVersion} yarn build`, {
  fatal: true,
});
shell.exec('yarn build:types', {
  fatal: true,
});
```

https://github.com/shelljs/shelljs#execcommand--options--callback

Reply:
You're right it's better and I tried it but I couldn't get it to work.
So I looked a the issues and it turns out it's a know issue to be [fixed in 0.9 (not yet released)](https://github.com/shelljs/shelljs/issues/981#issuecomment-626840798).
The workaround apparently is to set it globally, which I didn't want because shelljs is used by other stuff like ship.js and I wasn't sure of the impact there.

Reply:
wouldn't we want this to be set globally anyway?

Reply:
I don't think that would be a good idea. As of now, there are two projects depending on it, there might be more in the future (if we introduce new dependencies or some dependencies introduce it).
[Shipjs seems to rely on the fact that it can ignore errors](https://github.com/algolia/shipjs/blob/96d6ff22214849b7c0c665b3d104b00ded43b38a/packages/shipjs-lib/src/lib/git/getCommitBodies.js#L5), so there is a risk of enabling it globally break the release scripts. Nor safe now, or the future.


```sh
yarn why shelljs
yarn why v1.22.10
[1/4] 🤔  Why do we have the module "shelljs"...?
[2/4] 🚚  Initialising dependency graph...
warning Resolution field "algoliasearch@3.35.0" is incompatible with requested version "algoliasearch@^3.35.1"
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "shelljs@0.8.3"
info Has been hoisted to "shelljs"
info This module exists because it's specified in "devDependencies".
info Disk size without dependencies: "316KB"
info Disk size with unique dependencies: "452KB"
info Disk size with transitive dependencies: "1.43MB"
info Number of shared dependencies: 11
=> Found "shipjs-lib#shelljs@0.8.4"
info This module exists because "shipjs#shipjs-lib" depends on it.
info Disk size without dependencies: "316KB"
info Disk size with unique dependencies: "452KB"
info Disk size with transitive dependencies: "1.43MB"
info Number of shared dependencies: 11
=> Found "@storybook/core#shelljs@0.8.4"
info This module exists because "@storybook#html#@storybook#core" depends on it.
info Disk size without dependencies: "348KB"
info Disk size with unique dependencies: "484KB"
info Disk size with transitive dependencies: "1.46MB"
info Number of shared dependencies: 11
✨  Done in 1.08s.
```

Reply:
I don't understand what you're afraid of, those scripts don't happen at the same time or the same state no? Where are you getting that the settings are persisted outside of the script?

Reply:
```suggestion
const results = [
  shell.exec(`NODE_ENV=production VERSION=${newVersion} yarn build`),
  shell.exec('yarn build:types'),
];

if (results.some(result.code !== 0)) {
  shell.exit(1);
}
```

ok what about this instead then? still checking the return code manually, but leaves visual room for the script